File: api/src/RadioactivityApp/MeasuringStations/MeasuringStation.cs
namespace RadioactivityApp.MeasuringStations;

public sealed class MeasuringStation
{
    public string Kenn { get; init; } = "";
    public double Latitude { get; init; }
    public double Longitude { get; init; }
    public string? ZipCode { get; init; }
    public string? Name { get; init; }
}

File: api-client/lib/src/models/measuring_station.dart
import 'package:meta/meta.dart';

@sealed
@immutable
@internal
class MeasuringStation {
  final String kenn;
  final double latitude;
  final double longitude;
  final String? zipCode;
  final String? name;

  MeasuringStation({
    required this.kenn,
    required this.latitude,
    required this.longitude,
    this.zipCode,
    this.name,
  });

  MeasuringStation.fromJson(Map<String, dynamic> json)
      : kenn = json['Kenn'] as String,
        latitude = json['Latitude'] as double,
        longitude = json['Longitude'] as double,
        zipCode = json['ZipCode'] as String?,
        name = json['Name'] as String?;
}
